packages/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.ts

```diff
@@ -1137,11 +1121,23 @@ async function getFunctionArgsSuggestions(
       );
     }
 
+    const supportedFieldTypes = fnDefinition.signatures
+      .flatMap((signature) => {
+        if (signature.params.length > argIndex) {
+          return signature.params[argIndex].constantOnly ? '' : signature.params[argIndex].type;
+        }
+        if (signature.minParams) {
+          return signature.params[signature.params.length - 1].type;
+        }
+        return [];
+      })
+      .filter(Boolean);
```

Comment:
Use `nonNullable` to preserve types.

```suggestion
      .filter(nonNullable);
```

Reply:
done in https://github.com/elastic/kibana/pull/180509/commits/0b9c510e944973a8a9ca990e5678aad9bd62783e